hxd/fs/FileEntry.hx

```diff
@@ -1,34 +1,142 @@
 package hxd.fs;
 
-class FileEntry {
+/**
+	The base class representing a single file in the FileSystem.
 
+	Should not be instantiated directly, as specific implementation lies on the FileSystem type used.
+**/
+class FileEntry {
+	/**
+		The name of the file.
+	**/
 	public var name(default, null) : String;
+	/**
+		The local path of the file.
+	**/
 	public var path(get, never) : String;
+	/**
+		The directory path the file is placed in.
+	**/
 	public var directory(get, never) : String;
+	/**
+		File extension of the file without the dot (i.e. `png`).
+	**/
 	public var extension(get, never) : String;
+	/**
+		The size of the file in bytes.
+	**/
 	public var size(get, never) : Int;
+	/**
+		Whether this entry is a directory.
+
+		Depending on FileSystem used, it may contain directory entries.
+		In that case they can be iterated over, but should not be accessed as files.

```

Comment:
I don't understand the "Depending on FileSystem used". This is the specification of the filesystem, some sub file systems might not respect it but it should be documented in their code, not here.